mock_can_data.py

```diff
@@ -0,0 +1,68 @@
+# This script automatically generates and sends CAN messages based off of the DBC file
+# Ensure that you have can and cantools installed, which can be accomplished by running
+# pip install -r requirements.txt
+# Alternatively you could also just run
+# pip install can && pip install cantools
+
+# If you are running this script with virtual CAN ensure that it is set up first
+# You can run the below commands
+# sudo modprobe vcan
+# sudo ip link add dev vcan0 type vcan
+# sudo ip link set up vcan0
+
+import cantools
+import can
+import time
+import random
+
+# A negative value for num_messages will cause the script to send CAN messages forever
+num_messages = -1
+sleep_time_s = 1
+can_messages = []
+
+try:
+    db = cantools.database.load_file('system_can.dbc')
+except:
+    print("Must generate DBC file first")
+    print("Run make gen && make gen-dbc")
+
+# This can be edited depending on the CAN interface
+can_bus = can.interface.Bus('vcan0', bustype='socketcan')
+
+def main():
+    get_messages()
+    iterate_message_and_signal()
+
+def get_messages():
+    for msg in db.messages:
+        can_messages.append(msg)
+
+def iterate_message_and_signal():
+    num_messages_sent = 0
+    if num_messages < 0 :
+        while True:
+            try:
+                send_message()
+                num_messages_sent +=1
+                time.sleep(sleep_time_s)
+            except KeyboardInterrupt:
+                break
+    else:
+        while num_messages_sent < num_messages:
+            send_message()
+            num_messages_sent +=1
+            time.sleep(sleep_time_s)

```

Comment:
this is pretty messy... could you do something like
```
num_messages_sent = 0
to_continue = num_messages_sent < num_messages or num_messages < 0
while to_continue:
    ...
```
This way you shouldn't need to copy the code twice.
I don't think you need the `except KeyboardInterrupt` since doing `Ctrl+C` automatically kills the python script anyways.

Reply:
The keyboard interrupt is there so that it will exit the loop and display the total number of messages sent rather than just some other error

Reply:
makes sense 👍